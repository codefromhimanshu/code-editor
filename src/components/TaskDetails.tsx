
import React, { useState, useEffect } from 'react';
interface Task { id: string; title: string; description: string };

interface TaskDetailsProps {
  task: { id: string; title: string; description: string, status: string };
  onClose: () => void;
  onUpdate: (Task) => void;
  onDelete: (id: string) => void;
}

const TaskDetails: React.FC<TaskDetailsProps> = ({ task, onClose, onUpdate, onDelete }) => {

  const [title, setTitle] = useState(task.title);
  const [description, setDescription] = useState(task.description);
  const [status, setStatus] = useState(task.status);

  useEffect(() => {
    setTitle(task.title);
    setDescription(task.description);
    setStatus(task.status);
  }, [task])

  const handleUpdate = () => {
    console.log(task)
    onUpdate({...task, title, description, status});
  };

  const handleDelete = () => {
    console.log(task)
    onDelete(task.id);
  };

  return (
    <div className="absolute right-0 top-0 bottom-0 w-1/3 bg-white p-8 shadow-lg">
      {/* Close button */}
      <button onClick={onClose} className="mb-4 border p-2 rounded float-right">&times;</button>
      <div className="font-semibold mt-8 block">
        {/* Task title input */}
        <label htmlFor="title" className="font-semibold mb-2 block">Task Title</label>
        <input
          type="text"
          value={title}
          onChange={(e) => setTitle(e.target.value)}
          className="w-full mb-4 p-2 border rounded font-normal"
        />

        {/* Task description input */}
        <label htmlFor="description" className="font-semibold mb-2 block">Description</label>
        <textarea
          value={description}
          onChange={(e) => setDescription(e.target.value)}
          className="w-full mb-4 p-2 border rounded font-normal"
        />
        <label htmlFor="status" className="font-semibold mb-2 block">Status </label>
         <select 
            className=" mb-4 p-2 border rounded font-normal"
            onChange={(e) => setStatus(e.target.value)}
            value={status}
          >
            <option value="To Do">To Do</option>
            <option value="In Progress">In Progress</option>
            <option value="In Progress">In Review</option>
            <option value="Done">Done</option>
          </select>
          <div className='w-full'></div>
        {/* Update button */}
        <button onClick={handleUpdate} className="bg-blue-500 text-white px-4 py-2 rounded hover:bg-blue-600 mr-2">Update</button>
        <button onClick={handleDelete} className="bg-red-500 text-white px-4 py-2 rounded hover:bg-red-600">Delete</button>
      </div>
      
    </div>
  );
};

export default TaskDetails;
    