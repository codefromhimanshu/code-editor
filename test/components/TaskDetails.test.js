// tests/TaskDetails.test.tsx
import React from "react";
import renderer from "react-test-renderer";
import TaskDetails from "../../src/components/TaskDetails";

const mockTask = {
  id: "1",
  title: "Test Task",
  description: "Test Description",
  status: "To Do",
};

describe("TaskDetails", () => {
  it("renders correctly", () => {
    const component = renderer.create(
      <TaskDetails
        task={mockTask}
        onClose={() => {}}
        onUpdate={() => {}}
        onDelete={() => {}}
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  // More tests could be added here, such as testing interaction
});
