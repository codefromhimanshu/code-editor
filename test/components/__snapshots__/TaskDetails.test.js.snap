// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskDetails renders correctly 1`] = `
<div
  className="absolute right-0 top-0 bottom-0 w-1/3 bg-white p-8 shadow-lg"
>
  <button
    className="mb-4 border p-2 rounded float-right"
    onClick={[Function]}
  >
    Ã—
  </button>
  <div
    className="font-semibold mt-8 block"
  >
    <label
      className="font-semibold mb-2 block"
      htmlFor="title"
    >
      Task Title
    </label>
    <input
      className="w-full mb-4 p-2 border rounded font-normal"
      onChange={[Function]}
      type="text"
      value="Test Task"
    />
    <label
      className="font-semibold mb-2 block"
      htmlFor="description"
    >
      Description
    </label>
    <textarea
      className="w-full mb-4 p-2 border rounded font-normal"
      onChange={[Function]}
      value="Test Description"
    />
    <label
      className="font-semibold mb-2 block"
      htmlFor="status"
    >
      Status 
    </label>
    <select
      className=" mb-4 p-2 border rounded font-normal"
      onChange={[Function]}
      value="To Do"
    >
      <option
        value="To Do"
      >
        To Do
      </option>
      <option
        value="In Progress"
      >
        In Progress
      </option>
      <option
        value="In Progress"
      >
        In Review
      </option>
      <option
        value="Done"
      >
        Done
      </option>
    </select>
    <div
      className="w-full"
    />
    <button
      className="bg-blue-500 text-white px-4 py-2 rounded hover:bg-blue-600 mr-2"
      onClick={[Function]}
    >
      Update
    </button>
    <button
      className="bg-red-500 text-white px-4 py-2 rounded hover:bg-red-600"
      onClick={[Function]}
    >
      Delete
    </button>
  </div>
</div>
`;
